src/statemachine.rs

```diff
@@ -213,16 +236,21 @@ impl StateMachine {
 
         // Check if the combination of device-protection and request-options
         // are allowing for 'discouraged', meaning no auth required.
-        if cmd.can_skip_user_verification(info) {
+        if cmd.can_skip_user_verification(info, uv_req) {
             return Ok(PinUvAuthResult::NoAuthRequired);
         }
 
         // Device does not support any auth-method
-        if !pin_configured && !supports_uv {
-            // We'll send it to the device anyways, and let it error out (or magically work)
+        if (skip_uv || !supports_uv) && !supports_pin {
             return Ok(PinUvAuthResult::NoAuthTypeSupported);
         }
 
+        // Device supports PINs, but a PIN is not configured. Signal that we
+        // can complete the operation if the user sets a PIN first.
+        if (skip_uv || !supports_uv) && !pin_configured {
+            return Err(AuthenticatorError::PinError(PinError::PinNotSet));
```

Comment:
I'm also not 100% certain of this. We might not want to use the PinNotSet error code here. We might want one specific to this case.

Reply:
I don't really see it used anywhere else or for anything else, so I'm not sure this is a problem. But since this is `PinError` and not `CommandError::StatusCode`, we can easily add another variant for this. Fine with me either way.